### PR TITLE
More native wrappers

### PR DESCRIPTION
**Pull request by dpethes**
A couple of new or modified methods using native calls for ImGui class. This is mainly preparation work for porting of Memory editor sample.

### CODE REVIEW

**Comment by mellinoe** [comment]
Hey @dpethes , thanks for the submission! Just wanted to let you know that I'm not ignoring this and I will try to take a look later tonight.

**Comment by dpethes** [comment]
Cool! Also, the memory editor port is basically ready, if you're interested I can commit it as well. It needs some data to truly show something interesting, but I could generate something on the fly.

**Comment by mellinoe** [comment]
Very cool! Are you referring to this sample? https://gist.github.com/ocornut/0673e37e54aff644298b

**Comment by dpethes** [comment]
Yes, that's the one.

**Comment by mellinoe** [comment]
Apologies for the delay on this. I have been really busy with some other stuff but am going to try to look at this in an hour or so.

File: src/ImGui.NET/ImGui.cs [hunk 1]
```diff
@@ -896,7 +940,7 @@ public static void SameLine()
             ImGuiNative.igSameLine(0, 0);
         }
 
-        public static void SameLine(float localPositionX, float spacingW)
+        public static void SameLine(float localPositionX = 0, float spacingW = -1.0f)
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: I think we should either

* Remove the parameterless overload above this
or
* Make the parameterless overload use the same "default values" as here.

Thoughts?

Comment (dpethes) [inline src/ImGui.NET/ImGui.cs]: I'd rather remove the parameterless one - generally, I was re-adding the default param values that are missing from cimgui's interface.

File: src/ImGui.NET/ImGuiNative.cs [hunk 2]
```diff
@@ -272,7 +272,7 @@ public static unsafe class ImGuiNative
         public static extern void igTextWrapped(string fmt);
 
         [DllImport(cimguiLib)]
-        public static extern void igTextUnformatted(byte* text, byte* text_end);
+        public static extern void igTextUnformatted(string text, string text_end);
```
Comment (mellinoe) [inline src/ImGui.NET/ImGuiNative.cs]: I don't believe this is correct; these should be pointers to the beginning and end of the data, although the second parameter is allowed to be null. There's not really a way to pass the second parameter correctly if it is typed as a string. I think we should keep the PInvoke declaration the same, and in the `ImGui.TextUnformatted` method above, just pin the string and pass its address as the first parameter.

**Comment by mellinoe** [comment]
This looks mostly good, but I'd like to do the change to `igTextUnformatted` a little differently as I mentioned in the comment. Let me know how you feel about it.

File: src/ImGui.NET/ImGui.cs (hunk 1 above)
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: Yes, I'd agree with you there. Go ahead and remove it and just leave the one with default parameters.

File: src/ImGui.NET/ImGuiNative.cs (hunk 2 above)
Comment (dpethes) [inline src/ImGui.NET/ImGuiNative.cs]: You're right, I wasn't thinking about the case when one would want to print a substring that doesn't start at the first character. In addition it seems that this is wrong in some other functions as well - igGetIdStrRange, igPushIdStrRange?

Comment (dpethes) [inline src/ImGui.NET/ImGuiNative.cs]: Also, these range functions are pretty much useless in c#, maybe omit them completely?

**Comment by mellinoe** [comment]
Great, thanks for fixing that. As for the other "range" functions, I can fix them later. They aren't exactly "useless" in C#, just harder to use. I am actually already using the `igTextUnformatted` function with start and end addresses in another project, which is why it jumped out at me.

Thanks for the contribution!

**Comment by dpethes** [comment]
Thanks for merging! 
I see now, you use it with a special class instead of string. Haven't thought of that.

**Pull request merged**